base/sort.jl

```diff
@@ -440,6 +440,41 @@ Variant of [`sort!`](:func:`sort!`) that returns a sorted copy of `v` leaving `v
 """
 sort(v::AbstractVector; kws...) = sort!(copymutable(v); kws...)
 
+function sort{T<:Integer}(v::Vector{T})
+    n = length(v)
+    if n > 1
+        min, max = extrema(v)
+        rangelen = max - min + 1
+        if rangelen < div(n,2)
+            return sort_int_range(v, rangelen, min)
+        end
+    end
+    return sort!(copymutable(v))
+end
+
+# sort for vectors of few unique integers
+function sort_int_range{T<:Integer}(x::Vector{T}, rangelen, minval)
```

Comment:
Why not make an in-place `sort_int_range!` function instead, and call it from `sort!` rather than `sort`?